feature/radio-buttons

schinns opened the pull request:
Solution to https://github.com/revery-ui/revery/issues/245 and Radio Button Example

I implemented simplest solution possible. If we want to try a different implementation let me know!

![radio-button](https://user-images.githubusercontent.com/21044999/51702694-aba73a00-1fd1-11e9-99d5-dd882dc0ff48.gif)

Akin909 commented on examples/RadioButtonExample.re:
```diff
@@ -0,0 +1,53 @@
+    component(slots => {
+      let (radioVal, setRadioVal, _slots: React.Hooks.empty) = React.Hooks.state("Select a button!", slots);
+      <View
+        style=Style.[
```
Don't think its part of our CI yet but think @bryphe was trying to add `refmt` as part of the process, if you have it installed you can point it at your files and it will autoformat them.
Not a blocker I don't think but would be nice to have :+1:

schinns replied:
I just ran `refmt RadioButtonsExample.re` and I saw some output in the terminal, but there were no changes made to commit.

bryphe replied:
Ah yes it's a bit confusing - you have to actually run `refmt --in-place RadioButtonsExample.re` - otherwise it just writes the output to the console

schinns replied:
ahh ok got it! thanks!

Akin909 commented:
@schinns this is awesome :100: I left a couple of comments (minor ones :+1:)

schinns commented:
@Akin909 thanks for your review! Woops didn't mean to push up `.DS_Store`, I removed the file and added it to `.gitignore`! I also ran `refmt` on the files I created. It didn't make any changes. 🤷‍♀️(or maybe I didn't run `refmt` properly?)

bryphe commented:
Awesome @schinns , thank you for the contribution! Very cool 💯 

Sorry about the CI issue, I've kicked it off manually for now - going to open a ticket on Azure CI too though and see if we have something misconfigured. It seems like occasionally it just doesn't run for PRs.

bryphe commented:
Just was trying it out - looks awesome @schinns ! Thank you for your work on this.

I noticed the CI kicked off this time, but it looked like there were still some formatting issues. It might be due to different versions of `refmt` - we're using the `3.4.0` reason CLI tools on the build machine.

Hope you don't mind - I formatted and committed that change to your branch so it'd be get picked up here.

Otherwise, looks great! I'll bring this in once the build is green.

bryphe commented:
CI is green! 🎉 Thank you @schinns - really appreciate the contribution! Bringing in now.
